src/a-runtime-console/kubernetes/ui/pipeline/build-stage-view/build-stage-view.component.html

```diff
@@ -16,43 +16,70 @@
       </div>
     </div>
     <div class="pipeline-container">
-      <div class="pipeline" *ngIf="!build.pipelineStages.length">
-        <div class="pipeline-stage no-stages">
-          <div id="pipeine-stage-name-no-stages" class="pipeline-stage-name">No stages have started.</div>
-        </div>
-      </div>
-      <div class="pipeline" *ngIf="build.pipelineStages.length">
-        <div class="pipeline-stage " *ngFor='let stage of displayStages'>
-          <div column="" class="pipeline-stage-column">
-            <div id="pipeine-stage-name" class="pipeline-stage-name {{build.statusPhase}}"
-                 title="stage status: {{stage.status}}">
-              {{stage.name}} &nbsp;
-              <span class="open-service-icon" *ngIf="stage.serviceUrl">
-                <a target="deployment"
-                   class="external-service"
-                   [href]="stage.serviceUrl"
-                   title="Open this service in a separate browser tab">
-                  <i class="fa fa-external-link" aria-hidden="true"></i>
-                </a>
-              </span>
-            </div>
-            <pipeline-status [stage]="stage"></pipeline-status>
+      <ng-container *ngIf="!build.logURL && !build.pipelineStages.length; then noLogs"></ng-container>
```

Comment:
If you add an `else noStages` to the end it should allow you to remove the line.

Reply:
hmm, now that I read it more carefully I see that you could get to the `else` if `build.pipelineStages.length` has length and that is the 3rd line

If that is true then I would agree with you, you can't do the `else`

Reply:
I think I have it. Will this work?
```
<ng-container *ngIf="!build.pipelineStages.length; else pipelinesStarted">
    <ng-container *ngIf="!build.logURL; then noLogs; else noStages"></ng-container>
</ng-container>
```

You will need to try it to see if it works. Please make sure you have sufficient test coverage this. :)

Reply:
@joshuawilson thanks,  yes this should work. This was the reason i asked we need to nest it, do think this will be cleaner then 

`<ng-container *ngIf="!build.logURL && !build.pipelineStages.length; then noLogs"></ng-container>`
 `<ng-container *ngIf="build.logURL && !build.pipelineStages.length; then noStages"></ng-container>`
  `<ng-container *ngIf="build.pipelineStages.length; then pipelinesStarted"></ng-container>`

I thought this would be cleaner, if you feel other way i'll make the changes. And tests i have added covering these.

Reply:
@joshuawilson thanks for the suggestion it looks clean and tested it works fine.